File: 2021/october/day-4/solution1.cpp
class Solution {
    public:
        int islandPerimeter(vector<vector<int>>& grid) 
        {
            
            int result=0; 
            
            for(int i=0; i<grid.size(); i++)
            {
                for(int j=0; j<grid[0].size(); j++)
                {
                    if(grid[i][j])
                    {
                        result+=4;
                        
                        if(i>0 && grid[i-1][j]) 
                            result-=2;
                        if(j>0 && grid[i][j-1]) 
                            result-=2;
                    }
                }
            }
            return result;
        }
    };

File: 2021/november/day-4/solution1.py
class Solution:
        def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
            stack = deque([(root,False)])
            
            result = 0
            
            while stack:
                node,isleft = stack.popleft()
                if not node: continue
                if not (node.left or node.right) and isleft:
                    result+=node.val
                    continue
                stack.append((node.left,True))
                stack.append((node.right,False))
            
            return result